### PR TITLE
add VSQToken contract and test cases

### PR DESCRIPTION
**Pull request by Jianzhuo**
Relate: #4

### CODE REVIEW

File: contracts/Staking.sol [hunk 1]
```diff
@@ -437,6 +440,8 @@ contract Staking is IStaking, Initializable, OwnableUpgradeable, Constants {
         ISQToken(SQToken).burn(burnAmount);
         IERC20(SQToken).safeTransfer(msg.sender, availableAmount);
 
+        stakedAmount[msg.sender] -= amount;
```
Comment (HZHAIX) [inline contracts/Staking.sol:443]: This should in `_startUnbond` other than`_withdrawARequest`, be consistent with `totalStakingAmount`

File: test/VSQToken.test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        await staking.connect(delegator).delegate(indexer2.address, etherParse('5'));
+        expect(await token.balanceOf(indexer.address)).to.equal(etherParse('10'));
+        expect(await vtoken.balanceOf(indexer.address)).to.equal(etherParse('20'));
+        expect(await token.balanceOf(indexer2.address)).to.equal(etherParse('10'));
+        expect(await vtoken.balanceOf(indexer2.address)).to.equal(etherParse('20'));
+        expect(await token.balanceOf(delegator.address)).to.equal(etherParse('5'));
+        expect(await vtoken.balanceOf(delegator.address)).to.equal(etherParse('15'));
+
+        await staking.connect(delegator).undelegate(indexer.address, etherParse('2'));
+        await startNewEra(mockProvider, eraManager);
```
Comment (HZHAIX) [inline test/VSQToken.test.ts:49]: Check delegator vbalance before withdraw, should be 15 as well

File: contracts/Staking.sol (hunk 1 above)
Comment (HZHAIX) [reply contracts/Staking.sol:443]: During the locked period, the token is not `staked` as it is removed from total staking.

Comment (HZHAIX) [reply contracts/Staking.sol:443]: Can we change `stakedAmount` to `lockedAmount`, so lockedAmount include `stakedAmount` + `amount in locked period`. @ianhe8x @neo-sun

Comment (neo-sun) [reply contracts/Staking.sol:443]: yeah, lockedAmount will be better.

File: contracts/Staking.sol [hunk 3]
```diff
@@ -121,8 +121,9 @@ contract Staking is IStaking, Initializable, OwnableUpgradeable, Constants {
     mapping(address => uint256) public withdrawnLength;
     //active delegation from delegator to indexer, delegator->indexer->amount
     mapping(address => mapping(address => StakingAmount)) delegation;
-    //each delegator total staking amount, delegator->amount
-    mapping(address => uint256) public stakedAmount;
+    //each delegator total locked amount, delegator->amount
```
Comment (HZHAIX) [inline contracts/Staking.sol:124]: Remove this comment

**Pull request merged**